### ISSUE TITLE
mingw-w64-crt strange bug when building gcc

### ISSUE THREAD

**Issue by revelator** [msys2/MINGW-packages#15476]
### Description / Steps to reproduce the issue

build mingw-w64-gcc using makepkg-mingw.

### Expected behavior

should work.

### Actual behavior

throws an error now in the code for making gcc's output colored (fancy_abort).
reverting the crt and headers to previous version allows building it again.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MSYS_NT-10.0-19044

### MINGW environments affected

- [X] MINGW64
- [X] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by Biswa96** [comment]
Pleas provide the build error.

**Comment by revelator** [comment]
ill have to reinstall the crt and attempt building gcc with it again hang on...

**Comment by revelator** [comment]
```
../../../gcc-11.3.0/libcpp/system.h:404:30: error: expected identifier before string constant
  404 | #define abort() fancy_abort (__FILE__, __LINE__, __FUNCTION__)
      |                              ^~~~~~~~
../../../gcc-11.3.0/libcpp/system.h:404:30: error: expected ',' or '...' before string constant
../../../gcc-11.3.0/libcpp/system.h:404:30: error: expected identifier before string constant
  404 | #define abort() fancy_abort (__FILE__, __LINE__, __FUNCTION__)
      |                              ^~~~~~~~
../../../gcc-11.3.0/libcpp/system.h:404:30: error: expected ',' or '...' before string constant
```

same with gcc-12.2

**Comment by Biswa96** [comment]
Aren't those fixed in this commit https://github.com/msys2/MINGW-packages/commit/4360ed1a7470728be1dba0687df764604f1992d9 ? Have you tried to compile mingw-w64-gcc with those changes?

**Comment by revelator** [comment]
well i tried the gcc-12.2.0 pull from a day or two ago and it still throws the same error.

**Comment by revelator** [comment]
updated the gcc pull let me try again, might not have made it into my previous pull.

**Comment by revelator** [comment]
yep that fixed it.

**Comment by revelator** [comment]
well that brought it further but still no cigar it now fails here ->

```
../../gcc-11.3.0/gcc/c/gccspec.c
../../gcc-11.3.0/gcc/jit/jit-w32.c: In function 'char* gcc::jit::win_mkdtemp()':
../../gcc-11.3.0/gcc/jit/jit-w32.c:213:11: error: 'rand_s' was not declared in this scope; did you mean 'rand_num'?
  213 |       if (rand_s (&rand_num) != 0)
      |           ^~~~~~
      |           rand_num
../../gcc-11.3.0/gcc/jit/jit-w32.c:222:48: error: 'UINT_MAX' was not declared in this scope
  222 |       rand_num = ((double)rand_num / ((double) UINT_MAX + 1 ) * 1000000);
```

**Comment by revelator** [comment]
[0400-jit-modern.zip](https://github.com/msys2/MINGW-packages/files/10662273/0400-jit-modern.zip)

this might work, added climits and stdlib.h to jit for UINT_MAX and rand_s.

**Comment by Biswa96** [comment]
> well that brought it further but still no cigar it now fails here. 

That's not our concern 😝 It's gcc 11.3.0 and mingw gcc is 12.2.0.

```
$ which gcc; gcc --version | head -n 1
/ucrt64/bin/gcc
gcc.exe (Rev10, Built by MSYS2 project) 12.2.0
```

**Comment by revelator** [comment]
even so as far as i know the jit code has not changed since version 11.3.0 so it might also fail there in gcc-12.2.0 unless you allready tried a rebuild of gcc and if it works there :), thats the only reason i bring it up :P

**Comment by revelator** [comment]
patch fixed the last bug so :)

**Comment by revelator** [comment]
ok 12.2.0 is not affected by this bug in jit, not really sure how it gets around it since neither climits nor stdlib.h is included by jit but maybe it pulls the definitions from elsewhere ?. 

my patch might atleast be of use for those who want to use the previous versions of gcc though.
